### ISSUE TITLE
Any plans for supporting 4.1?

### ISSUE THREAD

**Issue by easyaspi314** [pelya/screen-dimmer-pixel-filter#7]
A significant number of users (7%) are still on Android 4.1, API 16. Is there any paarticular reason we are only going to 4.2?

We could also probably go down to Gingerbread with NotificationCompat.

I am working on some major improvements currently for my fork, and would like to know.

**Comment by pelya** [comment]
Yes, there was some API I was using, that was only on Android 4.2. I need
to check the code.

**Comment by easyaspi314** [comment]
Found it, thanks.

**Comment by easyaspi314** [comment]
OK so I think the code is `Display.getRealSize();`. 

I found a solution that _may_ work using reflection on 4.0-4.1 and the current method on 4.2+. 

I will test it this evening on an emulator.

**Comment by easyaspi314** [comment]
I also heard that **KitKat blocked overlays on the navbar completely** because it could be a DDoS. So that might be why we are getting the weird patterns on some devices. We are trying to squash it.

I will test some stuff.

**Comment by easyaspi314** [comment]
@pelya can you explain what exactly you are doing here?

``` java
        Point displaySize = new Point();
        windowManager.getDefaultDisplay().getRealSize(displaySize);
        Point windowSize = new Point();
        windowManager.getDefaultDisplay().getSize(windowSize);
        displaySize.x += displaySize.x - windowSize.x;
        displaySize.y += displaySize.y - windowSize.y;
```

**Comment by pelya** [comment]
That's an attempt to cover system navigation buttons, it works in recent
apps list on Nexus 7, but in most cases nav buttons cannot be drawn over.
On Apr 30, 2016 8:32 PM, "easyaspi314 (Devin)" notifications@github.com
wrote:

@pelya https://github.com/pelya can you explain what exactlynyou are
doing here?

```
    Point displaySize = new Point();
    windowManager.getDefaultDisplay().getRealSize(displaySize);
    Point windowSize = new Point();
    windowManager.getDefaultDisplay().getSize(windowSize);
    displaySize.x += displaySize.x - windowSize.x;
    displaySize.y += displaySize.y - windowSize.y;
```

**Comment by pelya** [comment]
Yes, for KitKat the app shows a dialog to make user allow it 'draw over
other apps' permission in the in system settings.
